source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+
+var onednngraph = onednngraph || {};
+var json = json || require('./json');
+var base = base || require('./base');
+
+onednngraph.ModelFactory = class {
+
+    match(context) {
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        if (extension === 'json') {
+            const obj = context.open('json');
+            if (obj && obj.version && obj.graph) {
+                return 'onednngraph.json';
+            }
+        }
+        return undefined;
+    }
+
+    open(context, match) {
+        return onednngraph.Metadata.open(context).then((metadata) => {
+            const createModel = (metadata, symbol, version) => {
+                return new onednngraph.Model(metadata, symbol, version);
+            };
+            switch (match) {
+                case 'onednngraph.json': {
+                    let symbol = null;
+                    try {
+                        symbol = context.open('json');
+                    }
+                    catch (error) {
+                        const message = error && error.message ? error.message : error.toString();
+                        throw new onednngraph.Error("Failed to load symbol entry (" + message.replace(/\.$/, '') + ').');
+                    }
+                    const version = symbol && symbol.version ? symbol.version : null;
+                    return createModel(metadata, symbol, version);
+                }
+                default: {
+                    throw new onednngraph.Error("Unsupported oneDNN Graph format '" + match + "'.");
+                }
+            }
+        });
+    }
+};
+
+onednngraph.Model = class {
+
+    constructor(metadata, symbol, version) {
+        if (!symbol) {
+            throw new onednngraph.Error('JSON symbol data not available.');
+        }
+        if (symbol) {
+            if (!Object.prototype.hasOwnProperty.call(symbol, 'graph')) {
+                throw new onednngraph.Error('JSON file does not contain an oneDNN Graph \'graph\' property.');
+            }
+            if (!Object.prototype.hasOwnProperty.call(symbol, 'version')) {
+                throw new onednngraph.Error('JSON file does not contain an oneDNN Graph \'version\' property.');
+            }
+        }
+        this._format = 'oneDNN Graph';
+        this._version = 'oneDNN' + (version ? ' v' + version : '');
+        this._license = 'Apache License 2.0';
```

Comment:
Remove. This should return the license of the model as defined by the model author.

Reply:
got it.

Reply:
Remove `this._version = 'oneDNN' + (version ? ' v' + version : '');` as well, is part of `this._format`.